.github/workflows/release.yml

```diff
@@ -0,0 +1,55 @@
+name: Release
+
+on:
+  push:
+    branches:
```

Comment:
I think `tags` part is enough for what you plan to do.

Reply:
The intent is to have this workflow cover both snapshots and releases: on a regular merge to master we'll fire this workflow and expect it to publish a snapshot (since the library version at that point is suffixed with `SNAPSHOT`), and on a new tag we'll fire it and expect it to publish a release.